src/middleware/authentication.js

```diff
@@ -0,0 +1,27 @@
+/* Validates the user-provided auth token */
+const authentication = (req, res, next) => {
+  const authHeader = req.headers.Authorization;
+
+  /* Overrides user role for local development and testing purposes */
+  if (process.env.NODE_ENV !== 'production') {
+    const { role = 'admin' } = req.query;
+    req.user = { role }; // TODO: Update the shape of this object to match Firebase's user object
```

Comment:
Here I is where we'll be getting the auth token from the query and verifying the token with `firebase-admin`

Reply:
Yeah, but you wouldn't want to put that logic in this `if` block since this section is only for local development and testing purposes.

You would place that logic in the `if` block starting with `if (authHeader)`

Reply:
ahh okay gotcha that makes sense